### PR TITLE
Add DAG bundles model

### PR DESCRIPTION
**Pull request by jedcunningham**
This adds the initial model and FK for DAG Bundles. For now it is left as nullable (and left empty), but once parsing changes are complete it will become required.

The approach is very similar to triggerer, in that you have a classpath and kwargs so a bundle entity can be created.

### CODE REVIEW

File: airflow/models/dagbundle.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    @provide_session
+    def get_all_dag_bundles(
+        cls, *, session: Session = NEW_SESSION
+    ) -> list[tuple[DagBundleModel, BaseDagBundle]]:
+        """
+        Get all DAG bundles.
+
+        :param session: A database session.
+        :return: list of DAG bundles.
+        """
```
Comment (dstandish) [inline airflow/models/dagbundle.py:64]: shuold we make it private? maybe all models will be private anyway in 3.0?

**Review by dstandish — APPROVED** [review]

File: airflow/models/dagbundle.py (hunk 1 above)
Comment (jedcunningham) [reply airflow/models/dagbundle.py:64]: They are private in 3 anyways :)

File: airflow/migrations/versions/0050_3_0_0_add_dagbundlemodel.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+branch_labels = None
+depends_on = None
+airflow_version = "3.0.0"
+
+
+def upgrade():
+    op.create_table(
+        "dag_bundle",
+        sa.Column("id", UUIDType(binary=False), nullable=False),
+        sa.Column("name", sa.String(length=200), nullable=False),
```
Comment (kaxil) [inline airflow/migrations/versions/0050_3_0_0_add_dagbundlemodel.py:46]: We should probably use `airflow.migrations.db_types.StringID` so `collation_args` are applied for free

Comment (jedcunningham) [reply airflow/migrations/versions/0050_3_0_0_add_dagbundlemodel.py:46]: Good call: #44678

**Pull request merged**